app/src/main/java/com/example/vybe/AddEdit/AddEditController.java

```diff
@@ -0,0 +1,173 @@
+package com.example.vybe.AddEdit;
+
+import android.graphics.Bitmap;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.request.target.CustomTarget;
+import com.bumptech.glide.request.transition.Transition;
+import com.example.vybe.Models.SocSit;
+import com.example.vybe.Models.Vibe;
+import com.example.vybe.Models.VibeEvent;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+
+import java.io.ByteArrayOutputStream;
+import java.util.HashMap;
+
+public class AddEditController {
+    private static final String TAG = "AddEditController";
+    private static AddEditController instance;
+
+    private AddEditVibeEventActivity activity;
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private FirebaseAuth mAuth = FirebaseAuth.getInstance();
+    private StorageReference storageRef = FirebaseStorage.getInstance().getReference();
+    private String vibeEventDBPath = "Users/" + mAuth.getCurrentUser().getUid() + "/VibeEvents";
+    private VibeEvent vibeEvent;
+
+    private AddEditController() {
+    }
+
+    private void setActivity(AddEditVibeEventActivity activity) {
+        this.activity = activity;
+    }
+
+    public static AddEditController getInstance(AddEditVibeEventActivity activity) {
+        if (instance == null)
+            instance = new AddEditController();
+
+        instance.setActivity(activity);
+        return instance;
+    }
+
+    public void editVibeEvent(VibeEvent vibeEvent) {
+        this.vibeEvent = vibeEvent;
+        Vibe vibe = vibeEvent.getVibe();
+        String reason = vibeEvent.getReason();
+        SocSit socSit = vibeEvent.getSocSit();
+        String imagePath = vibeEvent.getImage();
+        Double latitude = vibeEvent.getLatitude();
+        Double longitude = vibeEvent.getLongitude();
+        assert vibeEvent.getID() != null : "Editing a VibeEvent that doesn't have an ID";
+
+        activity.setVibe(vibe);
+        if (reason != null) activity.setReason(reason);
+        if (socSit != null) activity.setSocSit(socSit);
+        if (imagePath != null) {
+            loadAndSetImage(imagePath);
+        }
+    }
+
+    public void addVibeEvent() {
+        vibeEvent = new VibeEvent();
+
+        String id = db.collection(vibeEventDBPath).document().getId();
+        vibeEvent.setID(id);
+    }
+
+    public void saveVibeEvent(Vibe vibe, String reason, SocSit socSit, Bitmap image, Double latitude, Double longitude) {
+        vibeEvent.setOwner(mAuth.getCurrentUser().getDisplayName());
```

Comment:
I don't like this. Store `mAuth.getCurrentUser()` in a variable called like `currentUser` or something that can be accessed globally by this class